core/src/replay_stage.rs

```diff
@@ -1720,27 +1727,37 @@ impl ReplayStage {
         verify_recyclers: &VerifyRecyclers,
         log_messages_bytes_limit: Option<usize>,
         prioritization_fee_cache: &PrioritizationFeeCache,
+        replayer_handle: &ReplayerHandle,
     ) -> result::Result<usize, BlockstoreProcessorError> {
         let mut w_replay_stats = replay_stats.write().unwrap();
         let mut w_replay_progress = replay_progress.write().unwrap();
         let tx_count_before = w_replay_progress.num_txs;
-        // All errors must lead to marking the slot as dead, otherwise,
-        // the `check_slot_agrees_with_cluster()` called by `replay_active_banks()`
-        // will break!
-        blockstore_processor::confirm_slot(
-            blockstore,
-            bank,
-            &mut w_replay_stats,
-            &mut w_replay_progress,
-            false,
-            transaction_status_sender,
-            Some(replay_vote_sender),
-            None,
-            verify_recyclers,
-            false,
-            log_messages_bytes_limit,
-            prioritization_fee_cache,
-        )?;
+
+        let mut did_process_entries = true;
+
+        // more entries may have been received while replaying this slot.
+        // looping over this ensures that slots will be processed as fast as possible with the
+        // lowest latency.
+        while did_process_entries {
```

Comment:
Doesn't this give preference to the current bank over other banks?

Reply:
seems ideal? you nede to finish bank before you can vote and work on next one